### PR TITLE
Make global tools work with helix

### PR DESCRIPTION
**Pull request by ViktorHofer**
We specify DOTNET_ROOT as otherwise a global tool's shim runs against the globally installed SDK host. Default the global tools dir to the runtime directory of if ArchiveTests equals true.

### CODE REVIEW

File: src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props [hunk 1]
```diff
@@ -11,6 +11,12 @@
   <ItemGroup Condition="'$(BuildingNETFxVertical)' == 'true'">
     <RunScriptCommands Include="set DEVPATH=%RUNTIME_PATH%" />
   </ItemGroup>
+
+  <!-- Set DOTNET_ROOT to route global tools shims to the correct host. -->
+  <ItemGroup Condition="'$(BuildingNETCoreAppVertical)' == 'true'">
+    <RunScriptCommands Condition="'$(TargetOS)' == 'Windows_NT'" Include="set DOTNET_ROOT=$(RunScriptHostDir)" />
```
Comment (safern) [inline src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: No longer doing this in sendToHelix.proj? As a HelixPreCommand?

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: There we could do it and only do it if a Coverage flag is set to true.

Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: Currently in master if you run /t:RebuildAndTest /p:Coverage=true the globally installed dotnet host is used for the global tools shims which is wrong. Therefore we need it here.

File: src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets [hunk 2]
```diff
@@ -24,11 +23,15 @@
...
-    <RunScriptHostDir Condition="'$(BuildingNETCoreAppVertical)' == 'true' AND '$(TargetOS)' != 'Windows_NT'">$RUNTIME_PATH/</RunScriptHostDir>
-    <RunScriptHost Condition="'$(BuildingNETCoreAppVertical)' == 'true' AND '$(TargetOS)' == 'Windows_NT'">$(RunScriptHostDir)dotnet.exe</RunScriptHost>
-    <RunScriptHost Condition="'$(BuildingNETCoreAppVertical)' == 'true' AND '$(TargetOS)' != 'Windows_NT'">$(RunScriptHostDir)dotnet</RunScriptHost>
+  <PropertyGroup Condition="'$(BuildingNETCoreAppVertical)' == 'true'">
+    <RunScriptHostDir Condition="'$(TargetOS)' == 'Windows_NT'">%RUNTIME_PATH%\</RunScriptHostDir>
+    <RunScriptHostDir Condition="'$(TargetOS)' != 'Windows_NT'">$RUNTIME_PATH/</RunScriptHostDir>
+    <RunScriptHost Condition="'$(TargetOS)' == 'Windows_NT'">$(RunScriptHostDir)dotnet.exe</RunScriptHost>
+    <RunScriptHost Condition="'$(TargetOS)' != 'Windows_NT'">$(RunScriptHostDir)dotnet</RunScriptHost>
+    
+    <GlobalToolsDir Condition="'$(ArchiveTests)' == 'true'">$(RunScriptHostDir)</GlobalToolsDir>
```
Comment (safern) [inline src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: Why is this needed here?

Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: For helix we copy the global tool directory content into the runtime directory. Thus is ArchiveTests=true we use runtime directory as the global tools path.

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: Will this targets be available for sendtohelix.proj? I also see that we set GlobalToolsDir to point to artifacts\tools in Directory.Build.props

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: I might be missing some place where GlobaltoolsDir is being used in order to run code coverage.

Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: No this is not imported in sendtohelix.proj as that project isn't a test project. sendtohelix.proj copies the global tools dir which by default is under `artifacts/tools` into the runtime directory.

Changing the `GlobalToolsDir` variable here only affects the command line argument in the run script.

Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.CoreFxTesting/build/core/coverage/Coverage.targets#L7

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.targets]: Makes sense. Sorry I’m tired and not thinking straight. :cries:

File: src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props (hunk 1 above)
Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: Do we want to do this just when Coverage=true?

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: Do we want to do this just when Coverage=true?

Comment (safern) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: I was thinking more of just adding the tools to the helix runtime payload only for coverage to not make it bigger in size for all the cases. But if we will be using other tools then I’m fine with it.

Comment (ViktorHofer) [reply src/Microsoft.DotNet.CoreFxTesting/build/core/Core.props]: Sure we can do that but that's unrelated to this change. That would be done in the accompanying corefx PR.
